File: Data Structures & Algorithms/three-integer-sum/submission-7.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        vector<vector<int>> res;
        for(int i=0;i<nums.size();i++){
            if(i>0&&nums[i]==nums[i-1]){
                continue;
            }
            if(nums[i]>0)
                break;
            int l=i+1,r=nums.size()-1;
            while(l<r){
                int curr=nums[i]+nums[l]+nums[r];
                if(curr>0){
                    r=r-1;
                }else if(curr==0){
                    res.push_back({nums[i],nums[l],nums[r]});
                    l++;
                    r--;
                    while(l<r &&nums[l]==nums[l-1])
                        l++; 
                }else{
                    l=l+1;
                }
            }
                
            
        }
        return res;
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-7.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        res = 0
        l = r = 0

        while r < len(nums) - 1:
            further = 0
            for i in range(l,r+1):
                further=max(further,i+nums[i])
            res+=1
            l=r+1
            r=further
        return res